examples/column/fct_advection.jl

```diff
@@ -0,0 +1,118 @@
+using Test
+using LinearAlgebra
+using OrdinaryDiffEq: ODEProblem, solve, SSPRK33
+
+import ClimaCore:
+    Fields,
+    Domains,
+    Topologies,
+    Meshes,
+    DataLayouts,
+    Operators,
+    Geometry,
+    Spaces
+
+
+# Advection Equation, with constant advective velocity (so advection form = flux form)
+# ∂_t y + w ∂_z y  = 0
+# the solution translates to the right at speed w,
+# so at time t, the solution is y(z - w * t)
+
+# visualization artifacts
+ENV["GKSwstype"] = "nul"
+using ClimaCorePlots, Plots
+Plots.GRBackend()
+dir = "fct_advection"
+path = joinpath(@__DIR__, "output", dir)
+mkpath(path)
+
+function linkfig(figpath, alt = "")
+    # buildkite-agent upload figpath
+    # link figure in logs if we are running on CI
+    if get(ENV, "BUILDKITE", "") == "true"
+        artifact_url = "artifact://$figpath"
+        print("\033]1338;url='$(artifact_url)';alt='$(alt)'\a\n")
+    end
+end
+
+FT = Float64
+t₀ = FT(0)
+t₁ = FT(1)
+z₀ = FT(0)
+zₕ = FT(1)
+z₁ = FT(1)
+speed = FT(1.0)
+C = FT(0.5) # flux-correction coefficient: ∈ [0,1] with
+#                              0 = first-order upwinding
+#                              1 = third-order upwinding
+n = 2 .^ 6
+
+domain = Domains.IntervalDomain(
+    Geometry.ZPoint{FT}(-π),
+    Geometry.ZPoint{FT}(π);
+    boundary_names = (:bottom, :top),
+)
+mesh = Meshes.IntervalMesh(domain, nelems = n)
+cs = Spaces.CenterFiniteDifferenceSpace(mesh)
+fs = Spaces.FaceFiniteDifferenceSpace(cs)
+zc = Fields.coordinate_field(cs)
+
+# Unitary, constant advective velocity
+w = Geometry.WVector.(speed .* ones(FT, fs))
+
+# Define a pulse wave or square wave
+pulse(z, t, z₀, zₕ, z₁) = abs(z.z - speed * t) ≤ zₕ ? z₁ : z₀
+
+# Initial condition
+y0 = pulse.(zc, 0.0, z₀, zₕ, z₁)
+
+function f!(dydt, y, parameters, t)
```

Comment:
Let's move this function up, before `FT = Float64`, to ensure that we've removed all closures.